core/deployment/src/main/java/io/quarkus/deployment/QuarkusConfig.java

```diff
@@ -12,6 +12,10 @@
 import io.quarkus.builder.item.SimpleBuildItem;
 import io.quarkus.deployment.configuration.ConfigurationError;
 
+/**
+ * @deprecated Do not use this class anymore, instead try {@code ConfigProvider.getConfig.getValue()} instead.
```

Comment:
This is not in any way an official policy but with stuff like this I have been trying to include a deletion version in the deprecation message (e.g. this will be removed at some point after 1.1), otherwise stuff like this tends to hang around forever. We probably should have some kind of official policy around this.

Reply:
Once we move to Java 11, they have a `since` property where you can say the version in which it was originally deprecated, and `forRemoval` which signifies that the element is going to go away.

It would be nice if they had a version where it was going to be removed, but one advantage to using `since` instead is that deprecation policies may change but history doesn't, so at each release we can say "we're removing all the items that are deprecated for removal since 1.1" or whatever.

But yeah I agree we need a policy, and it ought to cover configuration properties as well.

Reply:
That policy should also include https://github.com/quarkusio/quarkus/pull/5596

Reply:
I'll open an issue.